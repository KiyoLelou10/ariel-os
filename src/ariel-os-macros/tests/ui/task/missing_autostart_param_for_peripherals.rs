#![no_main]
#![feature(impl_trait_in_assoc_type)]

// FAIL: the `autostart` parameter must be present when requesting peripherals
#[ariel_os::task(peripherals)]
async fn main(_foo: Bar) {}

struct Bar;
