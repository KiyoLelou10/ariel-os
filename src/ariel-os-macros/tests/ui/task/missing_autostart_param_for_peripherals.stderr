error: custom attribute panicked
 --> tests/ui/task/missing_autostart_param_for_peripherals.rs:5:1
  |
5 | #[ariel_os::task(peripherals)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: the task must be `autostart` to receive peripherals
